### PR TITLE
Removed data_model components in vignettes (Issue #170)

### LINKED ISSUE #170: Avoid discussing data_model component in documentation
Originally, a `dm` consisted of a `src`, a `tables` and a `data_model` component. This is no longer true, but still occurs in the documentation in some places.

What's the best way to frame this? Perhaps something like:

- a `dm` consists of a collection of tables
- it maintains metadata for these tables: table and column names, and primary and foreign keys that link the tables
- all tables in a `dm` must be "on the same source", i.e. all must be local data frames or reside on the same database

Search for "three", "data_model", and "component" in `*.Rmd` files.

### PR DESCRIPTION
**Pull request by bbecane**
Two code examples throw an error (could not find function "dm_from_src") when knitting the vignettes that contain them.

Please suggest a fix for the error and review my changes.

Closes #170.

### CODE REVIEW

**Comment by TSchiefer** [comment]
To rule out possible reasons for the error (could not find function "dm_from_src"): did you install the version of {dm} on this branch to your library before knitting the vignette? (seems surprising that it would not know the function, since it's exported in that branch)

File: vignettes/dm-filtering.Rmd [hunk 1]
```diff
@@ -51,21 +51,15 @@ The data comes from the US Bureau of Transportation Statistics, and is documente
 To start with our exploration, we have to create a `dm` object from the {nycflights13} data.
 The built-in `dm::cdm_nycflights13()` function takes care of this.
 
-A [data model object](https://krlmlr.github.io/dm/articles/dm-class-and-basic-operations.html#class-dm) contains both data and metadata.
+A [data model object](https://krlmlr.github.io/dm/articles/dm-class-and-basic-operations.html#class-dm) contains data from the source tables, and metadata about the tables.
 
-If you would like to create a `dm` from other tables, you can use the `new_dm()`, `dm()` or `as_dm()` functions.
+If you would like to create a `dm` object from tables other than the example data, you can use the `new_dm()`, `dm()` or `as_dm()` functions.
```
Comment (krlmlr) [inline vignettes/dm-filtering.Rmd]:
```suggestion
If you would like to create a `dm` object from tables other than the example data, you can use the `new_dm()`, `dm()` or `as_dm()` functions.
See `vignette("dm-setup")` for details.
```

File: vignettes/dm-filtering.Rmd [hunk 2]
```diff
@@ -51,21 +51,15 @@ The data comes from the US Bureau of Transportation Statistics, and is documente
...
 ```
 
-A `dm` object output with data and metadata consists of three elements: 
-
-1. the table sources
-2. the data model
-3. a review of active filter conditions
-
-The console output is colored for a clear output:
+The console output of the 'dm` object shows its data and metadata, and is colored for a clarity:
```
Comment (krlmlr) [inline vignettes/dm-filtering.Rmd]:
```suggestion
The console output of the 'dm` object shows its data and metadata, and is colored for clarity:
```

File: vignettes/dm-visualization.Rmd [hunk 3]
```diff
@@ -26,9 +26,8 @@ knit_print.grViz <- function(x, ...) {
 }
 ```
 
-
 Once you have all your primary keys set and all foreign key relations defined, a graphical representation of your data model offers a condensed view of the tables and the relationships between the tables.
-The following functions can be used to visualize the data model of a `dm` object:^[The functions in this section rely heavily on related functions of [{datamodelr}](https://github.com/bergant/datamodelr).]
+The following functions can be used to visualize the `dm` object:^[The functions in this section rely heavily on related functions of [{datamodelr}](https://github.com/bergant/datamodelr).]
```
Comment (krlmlr) [inline vignettes/dm-visualization.Rmd]:
```suggestion
The following functions can be used to visualize the `dm` object:^[The code for the functions in this section is borrowed from the [{datamodelr}](https://github.com/bergant/datamodelr) package.]
```

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**